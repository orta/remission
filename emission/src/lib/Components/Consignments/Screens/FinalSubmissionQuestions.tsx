import * as React from "react"

import { LayoutAnimation, NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import Button from "../../Buttons/FlatWhite"
import ConsignmentBG from "../Components/ConsignmentBG"
import { BodyText, LargeHeadline, Subtitle } from "../Typography"

import { ConsignmentSetup, SearchResult } from "../"
import Spinner from "../../Spinner"
import TODO from "../Components/ArtworkConsignmentTodo"
import { Form, Label, Row } from "../Components/FormElements"
import Text from "../Components/TextInput"
import Toggle from "../Components/Toggle"
import Artist from "./Artist"
import Provenance from "./Provenance"
import SelectFromPhotoLibrary from "./SelectFromPhotoLibrary"
import Welcome from "./Welcome"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route // this gets set by NavigatorIOS
  setup: ConsignmentSetup
  submitFinalSubmission: (setup: ConsignmentSetup) => void
}

const Loader = (p, c) => <Spinner style={{ flex: 1 }} />

export default class FinalSubmissionQuestions extends React.Component<Props, ConsignmentSetup> {
  constructor(props) {
    super(props)
    this.state = props.setup
  }

  submitWork = () => {
    this.setState({ state: "SUBMITTED" }, () => this.props.submitFinalSubmission(this.state))
  }

  updateEdition = () => {
    // React Native's Typings are wrong here, I want to pass in
    // no arguments.
    const animate = LayoutAnimation.easeInEaseOut as any
    animate()

    this.setState({
      editionInfo: this.state.editionInfo ? null : {},
    })
  }

  updateSigned = () => this.setState({ signed: !this.state.signed })
  updateCert = () => this.setState({ certificateOfAuth: !this.state.certificateOfAuth })

  updateEditionSize = text => this.setState({ editionInfo: { ...this.state.editionInfo, size: text } })
  updateEditionNumber = text =>
    this.setState({ editionInfo: { ...this.state.editionInfo, number: parseInt(text, 10) } })

  render() {
    return (
      <ConsignmentBG>
        <Form title="Answer a few questions about the work">
          <Row>
            <Label>Is this an edition?</Label>
            <Toggle selected={!!this.state.editionInfo} left="YES" right="NO" onPress={this.updateEdition} />
          </Row>

          {this.state.editionInfo
            ? <Row>
                <Text
                  text={{
                    placeholder: "Edition Size",
                    keyboardType: "phone-pad",
                    onChangeText: this.updateEditionSize,
                    value: this.state.editionInfo && this.state.editionInfo.size,
                  }}
                  style={{ margin: 10 }}
                />
                <Text
                  text={{
                    placeholder: "Edition Number",
                    onChangeText: this.updateEditionNumber,
                    value:
                      this.state.editionInfo &&
                      this.state.editionInfo.number &&
                      this.state.editionInfo.number.toString(),
                  }}
                  style={{ margin: 10 }}
                />
              </Row>
            : null}

          <Row>
            <Label>Is this work signed?</Label>
            <Toggle selected={this.state.signed} left="YES" right="NO" onPress={this.updateSigned} />
          </Row>

          <Row>
            <Label>Do you have a certificate of authenticity?</Label>
            <Toggle selected={this.state.certificateOfAuth} left="YES" right="NO" onPress={this.updateCert} />
          </Row>
          <Row style={{ justifyContent: "center" }}>
            <View style={{ height: 43, width: 320, marginTop: 20 }}>
              <Button text="SUBMIT TO ARTSY" onPress={this.submitWork} style={{ flex: 1 }} />
            </View>
          </Row>
        </Form>
      </ConsignmentBG>
    )
  }
}
