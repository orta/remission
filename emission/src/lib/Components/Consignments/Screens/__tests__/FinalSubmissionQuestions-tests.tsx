import "jest-snapshots-svg"
import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import FinalSubmissionQuestions from "../FinalSubmissionQuestions"

const nav = {} as any
const route = {} as any

it("Sets up the right view hierarchy", () => {
  const tree = renderer
    .create(<FinalSubmissionQuestions navigator={nav} route={route} setup={{}} submitFinalSubmission={() => ""} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("Shows 3 sets of questions when there's no edition info", () => {
  const tree = renderer
    .create(<FinalSubmissionQuestions navigator={nav} route={route} setup={{}} submitFinalSubmission={() => ""} />)
    .toJSON()
  expect(tree).toMatchSVGSnapshot(750, 1334)
})

it("Shows 3 sets of questions when there's no edition info on iPad", () => {
  const tree = renderer
    .create(<FinalSubmissionQuestions navigator={nav} route={route} setup={{}} submitFinalSubmission={() => ""} />)
    .toJSON()
  expect(tree).toMatchSVGSnapshot(1536, 2048)
})

it("Shows and additional 2 inputs when there's edition info", () => {
  const tree = renderer
    .create(
      <FinalSubmissionQuestions
        navigator={nav}
        route={route}
        setup={{ editionInfo: {} }}
        submitFinalSubmission={() => ""}
      />
    )
    .toJSON()
  expect(tree).toMatchSVGSnapshot(750, 1334)
})

describe("Updating state", () => {
  it("adds submitted before triggering final submission on submit", () => {
    const submit = jest.fn()

    const f = new FinalSubmissionQuestions({ navigator: nav, route, setup: {}, submitFinalSubmission: submit })
    f.setState = jest.fn()
    f.submitWork()
  })
})
