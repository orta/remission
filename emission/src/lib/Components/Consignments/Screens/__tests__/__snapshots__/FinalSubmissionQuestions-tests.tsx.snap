// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "height": "100%",
          "maxWidth": 540,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "paddingTop": 40,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "textAlign": "center",
                },
                Object {},
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            Answer a few questions about the work
          </Text>
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingVertical": 6,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#cccccc",
                      "fontSize": 20,
                      "marginBottom": 18,
                      "marginTop": 18,
                      "paddingLeft": 25,
                      "paddingRight": 25,
                      "textAlign": "center",
                    },
                    Object {
                      "flex": 1,
                      "paddingLeft": 10,
                      "textAlign": "left",
                    },
                    Object {
                      "fontFamily": "AGaramondPro-Regular",
                    },
                  ]
                }
              >
                Is this an edition?
              </Text>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "green",
                        "borderColor": "white",
                        "borderRadius": 16,
                        "borderWidth": 1,
                        "height": 34,
                        "width": 64,
                      },
                      Object {
                        "backgroundColor": "black",
                      },
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "marginLeft": 8,
                          "marginRight": 8,
                          "marginTop": 9,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      NO
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 6,
                          "position": "absolute",
                          "right": 6,
                          "top": 4,
                        },
                        Object {
                          "flexDirection": "row",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "black",
                            "borderColor": "white",
                            "borderRadius": 12,
                            "borderWidth": 1,
                            "height": 24,
                            "width": 24,
                          },
                          Object {
                            "borderColor": "white",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingVertical": 6,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#cccccc",
                      "fontSize": 20,
                      "marginBottom": 18,
                      "marginTop": 18,
                      "paddingLeft": 25,
                      "paddingRight": 25,
                      "textAlign": "center",
                    },
                    Object {
                      "flex": 1,
                      "paddingLeft": 10,
                      "textAlign": "left",
                    },
                    Object {
                      "fontFamily": "AGaramondPro-Regular",
                    },
                  ]
                }
              >
                Is this work signed?
              </Text>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "green",
                        "borderColor": "white",
                        "borderRadius": 16,
                        "borderWidth": 1,
                        "height": 34,
                        "width": 64,
                      },
                      Object {
                        "backgroundColor": "black",
                      },
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "marginLeft": 8,
                          "marginRight": 8,
                          "marginTop": 9,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      NO
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 6,
                          "position": "absolute",
                          "right": 6,
                          "top": 4,
                        },
                        Object {
                          "flexDirection": "row",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "black",
                            "borderColor": "white",
                            "borderRadius": 12,
                            "borderWidth": 1,
                            "height": 24,
                            "width": 24,
                          },
                          Object {
                            "borderColor": "white",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingVertical": 6,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#cccccc",
                      "fontSize": 20,
                      "marginBottom": 18,
                      "marginTop": 18,
                      "paddingLeft": 25,
                      "paddingRight": 25,
                      "textAlign": "center",
                    },
                    Object {
                      "flex": 1,
                      "paddingLeft": 10,
                      "textAlign": "left",
                    },
                    Object {
                      "fontFamily": "AGaramondPro-Regular",
                    },
                  ]
                }
              >
                Do you have a certificate of authenticity?
              </Text>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "green",
                        "borderColor": "white",
                        "borderRadius": 16,
                        "borderWidth": 1,
                        "height": 34,
                        "width": 64,
                      },
                      Object {
                        "backgroundColor": "black",
                      },
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "marginLeft": 8,
                          "marginRight": 8,
                          "marginTop": 9,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                            "fontSize": 11,
                            "letterSpacing": 0.75,
                            "textAlign": "center",
                            "width": 24,
                          },
                          Object {
                            "color": "white",
                          },
                        ]
                      }
                    >
                      NO
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 6,
                          "position": "absolute",
                          "right": 6,
                          "top": 4,
                        },
                        Object {
                          "flexDirection": "row",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "black",
                            "borderColor": "white",
                            "borderRadius": 12,
                            "borderWidth": 1,
                            "height": 24,
                            "width": 24,
                          },
                          Object {
                            "borderColor": "white",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingVertical": 6,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 43,
                    "marginTop": 20,
                    "width": 320,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(255, 255, 255, 1)",
                        "borderColor": "#e5e5e5",
                        "borderWidth": 1,
                        "flex": 1,
                        "justifyContent": "center",
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontSize": 12,
                          },
                          Object {
                            "color": "black",
                            "fontFamily": "AGaramondPro-Regular",
                            "opacity": 1,
                          },
                          Object {
                            "fontFamily": "Avant Garde Gothic ITCW01Dm",
                          },
                        ]
                      }
                    >
                      SUBMIT TO ARTSY
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
